### PR TITLE
fix downloader not continue for android

### PR DESCRIPTION
**Pull request by drelaptop**
https://github.com/cocos2d/cocos2d-x/pull/17761

download not continue occurs when the values:
`countOfMaxProcessingTasks(jsb.Downloader)` <= `cc.macro.DOWNLOAD_MAX_CONCURRENT`

this PR fix it

### CODE REVIEW

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxDownloader.java [hunk 1]
```diff
@@ -327,9 +339,6 @@ public void run() {
                     catch (URISyntaxException e) {
                         break;
                     }
-                    catch (NullPointerException e) {
-                        break;
```
Comment (dumganhar) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxDownloader.java]: Why not to catch NullPointerException?

Comment (drelaptop) [reply cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxDownloader.java]: this is sync the changes from cocos2d-x repo, I don't know this clearly yet. and I don't think it's a good behavior to do nothing after catch a Exception. we should print warn-level log at least.

Comment (dumganhar) [reply cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxDownloader.java]: Yep, log output is better. But if we don't know the exact reason of removing `catch Null Pointer Exception` line, let's revert it at least. 
I agree that adding more detailed log output in NullPointerException catching block.

Comment (drelaptop) [reply cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxDownloader.java]: thanks for advice, I will research the reason

Comment (drelaptop) [reply cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxDownloader.java]: See the comments, `NullPointerException` will throw  when the URL is null
```java
     * @throws  NullPointerException
     *          If {@code str} is {@code null}
    public URI(String str) throws URISyntaxException {
        new Parser(str).parse(false);
    }
```
in  other side, before call those java code, the native code checked the URL
```C++
    std::shared_ptr<const DownloadTask> Downloader::createDownloadFileTask(const std::string& srcUrl,
...
    {
        do
        {
            ...
            if (0 == srcUrl.length() || 0 == storagePath.length())
            {
                if (onTaskError)
                {
                    onTaskError(*task, DownloadTask::ERROR_INVALID_PARAMS, 0, "URL or storage path is empty.");
                }
                task.reset();
                break;
            }
...
```
So, the `NullPointerException` will never occur.

**Pull request merged**